apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -25,6 +25,7 @@ import {
   PosixModeBits,
   JsonObject
 } from '@rushstack/node-core-library';
+import * as isCI from 'is-ci';
```

Comment:
Why is a CI machine special?  

And if it is special, should we be relying on [arbitrary heuristics](https://github.com/watson/ci-info/blob/f43f6a1cefff47fb361c88cf4b943fdbcaafe540/index.js#L17) to detect it? Seems like it would be better to explicitly specify it via an environment variable (`RUSH_CI_MACHINE=true` or somesuch).  For example, what if a developer wants to reproduce the CI behavior?

My inclination would be for Rush commands to behave identically on a CI machine and developer machine.

Reply:
> Why is a CI machine special?

From https://github.com/microsoft/rushstack/pull/1490#discussion_r320528269:
> "I'm slightly nervous about this change. This code was introduced when we were using NPM, whose cache could easily get into a corrupted state. Yes, you can fix it by running rush purge. But what happens when you have 20 lab machines that each has 10 build folders, and nobody has remote access to the machine? If a folder gets corrupted, all subsequent builds will fail for that agent. Somehow you have to figure out which machines/folders are corrupted, and manually run a job to clean them up. In that situation, it's much better to waste CPU time with a cache miss, versus wasting human time investigating an issue that causes downtime."

On a dev machine, a developer can simply run `rush purge` if the store is corrupted somehow.

> And if it is special, should we be relying on arbitrary heuristics to detect it? Seems like it would be better to explicitly specify it via an environment variable (RUSH_CI_MACHINE=true or somesuch). For example, what if a developer wants to reproduce the CI behavior?
> 
Just define the environment variable `CI` and `is-ci` will return true. This is used by many packages, including `pnpm` and `yarn`.

> My inclination would be for Rush commands to behave identically on a CI machine and developer machine.
There is good reason to automatically delete the store on CI machines. There is good reason to not automatically delete the store on dev machines. I don't see why we shouldn't implement this.

Reply:
> Just define the environment variable `CI` and `is-ci` will return true. 

Okay but now suppose we suspect the CI mode is causing trouble, and we want to turn it off?  Seems like `is-ci` provides no straightforward way to do that. `CI=0`?  `CI=false`? Neither of these seems to work. (?)

Fundamentally it feels like overengineering to implement a never-ending list of heuristics that try to detect every possible CI environment.  If we want Rush to behave a special way on CI machines, why not designate exactly one environment variable that enables the behavior, and report an error if the value is neither `'true` nor `false`/unset?

Magical heuristics avoid some effort and are cool only when things are working. But when something goes wrong, magic can be downright frustrating to work with.  (Is this  machine running in CI mode or not?  Well, here's the environment variables, but now we need to think through the 20 different logical tests that `is-ci` performs.  And which version of `is-ci` is it?  Ohhhh I figured it out, `is-ci` is returning true because it detected `SEMAPHORE=CandyCrush.exe`... apparently there's a CI system called "Semaphore"??)

> This is used by many packages, including `pnpm` and `yarn`.

We could implement a Rush feature to disable that somehow... 😎

I apologize for being opinionated about magical heuristics. Take it as just one person's opinion. I'm open-minded about it if other people see things differently.

@iclanton Was your concern with the behavior being different on CI machines?  (I'm actually okay with that part.)  Or was your concern with the `is-ci` heuristics?

Reply:
> If we want Rush to behave a special way on CI machines, why not designate exactly one environment variable that enables the behavior, and report an error if the value is neither 'true nor false/unset?

While that will work, it will also require all rush users to define this environment variable to get the following behavior:
1) "rush install" fails. Don't automatically delete the store. Deleting the store automatically would mean that *all* packages will need to be redownloaded when "rush install" is run again. Now if the store is in a corrupted state for some reason, then run "rush purge" can be run explicitly to delete the store. 
2) "rush install" fails on a CI machine. We don't want to ever leave the store in a potential bad state, so automatically delete the store.

> Magical heuristics
These predefined environment variables are actually well-documented, see https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables

Reply:
I'd be a lot more comfortable with this change if:

1. Rush clearly printed on startup that it is running in "CI Mode" and why it decided to run in that mode
2. We used our own `RUSH_CI_MODE=true` env variable.

Reply:
1. Rush now prints out the mode (CI mode vs NON-CI mode) as a part of Rush banner
2. The mode can be overridden using the environment variable `RUSH_CI_MODE`. 
  (a) Set to `CI` to force CI mode.
  (2) Set to `NON-CI` to force NON-CI mode.